pkg/controller/node/status.go

```diff
@@ -153,32 +171,32 @@ func getReadyMachines(currentConfig string, nodes []*corev1.Node) []*corev1.Node
 	return ready
 }
 
-func isNodeReady(node *corev1.Node) bool {
+func checkNodeReady(node *corev1.Node) error {
 	for i := range node.Status.Conditions {
 		cond := &node.Status.Conditions[i]
 		// We consider the node for scheduling only when its:
 		// - NodeReady condition status is ConditionTrue,
 		// - NodeOutOfDisk condition status is ConditionFalse,
 		// - NodeNetworkUnavailable condition status is ConditionFalse.
 		if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
-			glog.Infof("Node %s is reporting NotReady", node.Name)
-			return false
+			return fmt.Errorf("Node %s is reporting NotReady", node.Name)
 		}
 		if cond.Type == corev1.NodeOutOfDisk && cond.Status != corev1.ConditionFalse {
-			glog.Infof("Node %s is reporting OutOfDisk", node.Name)
-			return false
+			return fmt.Errorf("Node %s is reporting OutOfDisk", node.Name)
 		}
 		if cond.Type == corev1.NodeNetworkUnavailable && cond.Status != corev1.ConditionFalse {
-			glog.Infof("Node %s is reporting NetworkUnavailable", node.Name)
-			return false
+			return fmt.Errorf("Node %s is reporting NetworkUnavailable", node.Name)
 		}
 	}
 	// Ignore nodes that are marked unschedulable
 	if node.Spec.Unschedulable {
-		glog.Infof("Node %s is reporting Unschedulable", node.Name)
-		return false
+		return fmt.Errorf("Node %s is reporting Unschedulable", node.Name)
 	}
-	return true
+	return nil
+}
+
+func isNodeReady(node *corev1.Node) bool {
```

Comment:
couldn't we just return bool, error in checkNodeReady()?

Reply:
Yeah, though keeping the API the same involved less code churn for the callers.